File: src/slim.rb
require 'opal'
require 'opal-parser'

require 'thread'
require 'slim'

def slimBuilder(template, opts)
  version = (defined? Slim::VERSION) ? Slim::VERSION : Slim.version

  if version < '0.7.0'
    return Slim::Engine.new(template).render
  else
    return Slim::Template.new(opts){ template }.render(opts[:scope] || Hash.new)
  end
end

File: src/slim.js
require('opal-webpack!./slim.rb')
hashify = require('../lib/js/hashify');

self.onmessage = function(e) {
  var data = JSON.parse(e.data);

  var result;
  var err;
  try {
    var opts = data.options || {}
    result = Opal.Opal.$slimBuilder(data.html, hashify(opts))
  } catch (e) {
    err = JSON.stringify(e, Object.getOwnPropertyNames(e))
  }

  postMessage(JSON.stringify({html: result, err: err}))
}
